Sources/UserDefaultsGeneratorCore/Generator.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Generator.swift
+//  CYaml
+//
+//  Created by Yudai Hirose on 2019/09/18.
+//
+
+import Foundation
+
+public protocol Generator {
+    func generate() throws
+}
+
+public struct GeneratorImpl: Generator {
+    let configurations: [Configuration]
+    let outputPath: URL
+    public init(configurations: [Configuration], outputPath: URL) {
+        self.configurations = configurations
+        self.outputPath = outputPath
+    }
+    public func generate() throws {
+        let content = enumDefinition() + "\n" + userDefaultsExtensions()
+        try content.write(to: outputPath, atomically: true, encoding: .utf8)
+    }
+    
+    func enumDefinition() -> String {
+        let caseMap: (Configuration) -> String = { configuration in
+            switch configuration.key {
+            case nil:
+                return "case " + configuration.name
+            case let key?:
+                return "case " + configuration.name + " = " + "\"\(key)\""
+            }
+        }
+        return configurations
+            .grouped { $0.type }
+            .map { grouped in
+                let values = grouped.value.map(caseMap).joined(separator: "\n")
+                return """
+                public enum UDG\(grouped.key.rawValue)Key: String {
+                \(values)
+                }
+                """
+            }
+            .joined(separator: "\n")
+    }
+    
+    func userDefaultsExtensions() -> String {
+        return configurations
+            .grouped { $0.type }
+            .map { grouped in
+                let key = grouped.key
+                return """
+                extension UserDefaults {
+                    public func \(key.rawValue)(forKey key: UDG\(key.rawValue)Key) -> \(key.rawValue) {
+                        return \(key.getterMethodName)(forKey: key.rawValue)
+                    }
+                    public func set(_ value: \(key.rawValue), forKey key: UDG\(key.rawValue)Key) {
+                        set(value, forKey: key.rawValue)
+                        synchronize()
+                    }
+                }
+                """
+            }
+            .joined(separator: "\n")
+    }
+}
+
+extension Array {
+    func grouped<T: Comparable>(_ closure: (Element) -> (T)) -> [T: [Element]] {
```

Comment:
It's storage 🤔 
I think, `T` have to be `Hashable`. `Comparable` could not be dictionary key.

```suggestion
    func grouped<T: Hashable>(_ closure: (Element) -> (T)) -> [T: [Element]] {
```

Reply:
Yeah, that's right. I will changed it.

Write as a NOTE about using `Comparable`, the function tried to use [(T, [Element])] instead of [T: [Element]] to keep the elements order first. But it stopped in the middle. Because there was no time when 30 minutes limited live coding. 

The reason for to choise `[(T, [Element])] ` is that the change to `UserDefaults.generated.swift` maybe come out a lot when generate it every time.

Reply:
:done: